web-client/dist/nodejs/worker.js

```diff
@@ -9,6 +9,24 @@ const { Client } = require('./worker-wasm/index.js');
 
 // Provide a global WebSocket implementation, which is expected by the WASM code built for browsers.
 global.WebSocket = w3cwebsocket;
+// Workaround for libp2p's websocket-websys transport's worker detection,
+// which checks for web workers' `WorkerGlobalScope` interface.
```

Comment:
Per our discussion it seems like this is more a workaround to `nodejs` not supporting the `web-api` rather than a workaround to libp2p's websocket-websys transport's worker detection.

Reply:
As an alternative to the `WorkerGlobalScope` workaround, we could also look into using https://github.com/developit/web-worker to polyfill NodeJS's `worker_threads` to become Web Workers.

Reply:
But if the workaround necessary for libp2p is _only_ this assignment, then a whole other dependency seems like overkill.

But it seems your review comment is only about the describing code comment, not the mocking of `WorkerGlobalScope`, right?

Reply:
> But if the workaround necessary for libp2p is _only_ this assignment, then a whole other dependency seems like overkill.
> 
> But it seems your review comment is only about the describing code comment, not the mocking of `WorkerGlobalScope` itself, right?

Exactly